faker/providers/date_time/__init__.py

```diff
@@ -289,46 +292,53 @@ def date_time_ad(self, tzinfo=None):
         #       https://bugs.python.org/issue30684
         return datetime(1970, 1, 1,tzinfo=tzinfo) + timedelta(seconds=ts)
 
-    def iso8601(self, tzinfo=None):
+    def iso8601(self, tzinfo=None, end_datetime = None):
         """
         :param tzinfo: timezone, instance of datetime.tzinfo subclass
         :example '2003-10-21T16:05:52+0000'
         """
-        return self.date_time(tzinfo).isoformat()
+        return self.date_time(tzinfo,end_datetime=end_datetime).isoformat()
 
-    def date(self, pattern='%Y-%m-%d'):
+    def date(self, pattern='%Y-%m-%d', end_datetime = None):
         """
         Get a date string between January 1, 1970 and now
         :param pattern format
         :example '2008-11-27'
         """
-        return self.date_time().strftime(pattern)
+        return self.date_time(end_datetime=end_datetime).strftime(pattern)
 
-    def date_object(self):
+    def date_object(self, end_datetime = None):
         """
         Get a date object between January 1, 1970 and now
         :example datetime.date(2016, 9, 20)
         """
-        return self.date_time().date()
+        return self.date_time(end_datetime=end_datetime).date()
 
-    def time(self, pattern='%H:%M:%S'):
+    def time(self, pattern='%H:%M:%S', end_datetime = None):
         """
         Get a time string (24h format by default)
         :param pattern format
         :example '15:02:34'
         """
-        return self.date_time().time().strftime(pattern)
+        return self.date_time(end_datetime=end_datetime).time().strftime(pattern)
 
-    def time_object(self):
+    def time_object(self, end_datetime = None):
         """
         Get a time object
         :example datetime.time(15, 56, 56, 772876)
         """
-        return self.date_time().time()
+        return self.date_time(end_datetime=end_datetime).time()
+    
+    @classmethod
+    def _parse_end_datetime(self, value):
```

Comment:
`@classmethod`s shuold use `cls` instead of `self`

Reply:
Yeah, I thought I had changed it back after I realized this. Ill take care of it